src/libcore/array.rs

```diff
@@ -406,6 +408,73 @@ mod impls_using_const_generics {
         }
     }
 
+    #[unstable(feature = "array_value_iter", issue = "0")]
+    impl<T, const N: usize> IntoIterator for [T; N]
+    where
+        [T; N]: LengthAtMost32,
+    {
+        type Item = T;
+        type IntoIter = IntoIter<T, {N}>;
+
+        fn into_iter(self) -> Self::IntoIter {
+            IntoIter {
+                data: ManuallyDrop::new(self),
+                pos: 0,
+            }
+        }
+    }
+
+    /// A by-value array iterator.
+    ///
+    /// This struct is created by the `IntoIterator::into_iter`] method on
+    /// arrays.
+    #[unstable(feature = "array_value_iter", issue = "0")]
+    pub struct IntoIter<T, const N: usize> {
```

Comment:
I would like to see the constraint `[T; N]: LengthAtMost32,` repeated here, on `Drop`, and other impls, to make sure that const generics are not stably exposed (and this also means that `struct IntoIter<..>` itself cannot be stabilized until const generics are with this formulation).

Reply:
Also please extend the UI tests in Scott's fashion (consult https://github.com/rust-lang/rust/pull/62435#discussion_r300849161).